iai-callgrind-runner/tests/test_tools.rs

```diff
@@ -0,0 +1,44 @@
+use std::path::PathBuf;
+
+use iai_callgrind_runner::runner::costs::Costs;
+use iai_callgrind_runner::runner::dhat::logfile_parser::DhatLogfileParser;
+use iai_callgrind_runner::runner::tool::logfile_parser::{LogfileParser, LogfileSummary};
+fn dummy_summary(cmd: &str, pid: i32, cost_name: &str, cost: u64) -> LogfileSummary {
+    let costs = Costs::with_event_kinds([(cost_name.to_string(), cost)]);
+    LogfileSummary {
+        command: cmd.parse().unwrap(),
+        pid,
+        parent_pid: None,
+        fields: vec![],
+        details: vec![],
+        error_summary: None,
+        costs: Some(costs),
+        log_path: PathBuf::new(),
+    }
+}
+
+#[test]
+fn test_dhat_merge_logfile_summaries() {
+    let s1 = || dummy_summary("cmd1", 1, "cost", 1);
+    let s1b = || dummy_summary("cmd1", 1, "cost", 2);
+    let s2 = || dummy_summary("cmd2", 2, "cost", 1);
+    let dhat = DhatLogfileParser {
+        root_dir: PathBuf::new(),
+    };
+    assert_eq!(
+        dhat.merge_logfile_summaries(vec![s1b()], vec![s1()]),
+        vec![s1().merge(&s1b())]
```

Comment:
I don't think you're testing much here. In an it-test, I would have expected that the the resulting `ToolRunSummary`'s are created directly so the `LogfileSummary::merge`, ... methods are also tested. For example, the error with `old.pid` above would have become obvious that way.